Assets/_BForBoss/_Weapons/Scripts/EquipmentBehaviour.cs

```diff
@@ -0,0 +1,162 @@
+using System;
+using PerigonGames;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace Perigon.Weapons
+{
+    public class EquipmentBehaviour : MonoBehaviour
+    {
+        [SerializeField] private InputActionAsset _inputActions;
+        [SerializeField] private WeaponBehaviour[] _weapons = null;
+        private int _currentWeaponIndex = 0;
+
+        private InputAction _reloadInputAction = null;
+        private InputAction _fireInputAction = null;
+        private InputAction _swapWeaponInputAction = null;
+        private bool _isMouseScrollEnabled = true;
+        
+        public void Initialize()
+        {
+            EnableEquipmentPlayerInput();
+            SetupWeapons();
+            _swapWeaponInputAction.started += OnControllerSwapWeaponAction;
+        }
+
+        private void EnableEquipmentPlayerInput()
+        {
+            if (_reloadInputAction != null)
+            {
+                _reloadInputAction.Enable();
+            }
+
+            if (_fireInputAction != null)
+            {
+                _fireInputAction.Enable();
+            }
+
+            if (_swapWeaponInputAction != null)
+            {
+                _swapWeaponInputAction.Enable();
+            }
+
+            _isMouseScrollEnabled = true;
+        }
+        
+        private void DisableEquipmentPlayerInput()
+        {
+            if (_reloadInputAction != null)
+            {
+                _reloadInputAction.Disable();
+            }
+
+            if (_fireInputAction != null)
+            {
+                _fireInputAction.Disable();
+            }
+            
+            if (_swapWeaponInputAction != null)
+            {
+                _swapWeaponInputAction.Disable();
+            }
```

Comment:
Real minor thing but could we replace the enable/disable setup with null propagation? Just seems unnecessarily large as it is